lib/commanded/event_store/adapters/spear/event_publisher.ex

```diff
@@ -82,20 +82,45 @@ defmodule Commanded.EventStore.Adapters.Spear.EventPublisher do
     {:noreply, state, {:continue, :subscribe}}
   end
 
-  defp process_push(%Spear.Event{type: "$>"}, _) do
-    # This is a workaround to skip system events.
-    # Uisng a filter would be better, but for some reason the filter causes no events to be received.
-    :ok
-  end
-
-  defp process_push(push, %State{
+  defp process_push(%Spear.Event{} = event, %State{
          serializer: serializer,
          stream_prefix: stream_prefix,
-         pubsub: pubsub
+         pubsub: pubsub,
+         stream_name: listening_stream
        }) do
-    push
-    |> Mapper.to_recorded_event(serializer, stream_prefix)
-    |> publish(pubsub)
+    # This is a workaround to skip system events.
+    # Using a filter would be better, but for some reason the filter causes no events to be received.
+
+    action =
```

Comment:
this could be a tiny bit more idiomatic by using pattern matching, we could add a function `maybe_process_event/2` with event and listening_stream as parameters.
 
The two clauses that return `:ignore` could have different logs.
Wdyt?

Reply:
Done